scripts/core/ui/components/SelectUser.tsx

```diff
@@ -119,11 +121,16 @@ export class SelectUser extends SuperdeskReactComponent<IPropsSelectUser, IState
                     this.abortController?.abort();
                     this.abortController = new AbortController();
 
-                    const query = JSON.stringify(
-                        searchString != null && searchString.length > 0
-                            ? getUserSearchMongoQuery(searchString)
-                            : {},
-                    );
+                    const desk = sdApi.desks.getDeskById(this.props.deskId);
+                    const deskMemberIds = (desk?.members ?? []).map((member) => member.user);
+                    const query = {
+                        $and: [
+                            this.props.deskId != null ? {_id: {$in: deskMemberIds}} : {},
+                            searchString != null && searchString.length > 0
```

Comment:
if `searchString` is empty, return early and don't query at all.

Also adjust how query is made so you never end up with empty objects inside the query. Use `Array.push` to add things conditionally.

Reply:
I've tested it and it doesn't make any difference, do you think it has some other impact?

Reply:
Also if you want to do it this way can you give me a better example? If I push `{"$and":[]}` server responds with a 500. How do you suggest I implement this so that I don't need to check for `query.$and.length` and send the query conditionally?

Reply:
```
const desk = sdApi.desks.getDeskById(this.props.deskId);
const deskMemberIds = (desk?.members ?? []).map((member) => member.user);

if ((searchString ?? '').length < 1) {
    return Promise.resolve([]);
}

const query = {
    $and: [
        getUserSearchMongoQuery(searchString),
    ],
};

if (this.props.deskId != null) {
    query.$and.push({_id: {$in: deskMemberIds}});
}
```

Also try using `prepareSuperdeskQuery` to have it type safe.